File: frontend/src/services/khataService.js
import apiClient from './apiClient';

export const khataService = {
  addTransaction: async (khataData) => {
    const response = await apiClient.post('/khata', khataData);
    return response.data;
  },

  getWorkerKhata: async (workerId) => {
    const response = await apiClient.get(`/khata/worker/${workerId}`);
    return response.data;
  },

  getEmployerKhata: async (employerId) => {
    const response = await apiClient.get(`/khata/employer/${employerId}`);
    return response.data;
  },

  updateStatus: async (id, status) => {
    const response = await apiClient.patch(`/khata/${id}/status`, null, { params: { status } });
    return response.data;
  }
};

export default khataService;

File: backend/src/main/java/com/kaamchahiye/app/service/KhataService.java
package com.kaamchahiye.app.service;

import com.kaamchahiye.app.dto.request.KhataRequest;
import com.kaamchahiye.app.dto.response.KhataTransactionDto;
import java.util.List;

public interface KhataService {
    KhataTransactionDto addTransaction(KhataRequest request);
    List<KhataTransactionDto> getWorkerKhata(Long workerId);
    List<KhataTransactionDto> getEmployerKhata(Long employerId);
    KhataTransactionDto updateStatus(Long id, String status);
}
